### PR TITLE
Rule 4.1.2: Numbers of a float type should not be directly compared

### LINKED ISSUE #290: Rule 4.1.2: Numbers of a float type should not be directly compared with equality operator (==) or other methods like compareTo and equals.
Since floating-point numbers have precision problems in computer representation - as it was recommended in
Rule 4.1.1 - better to use BigDecimal instead when you need to make accurate computations and comparison.
The following code describes these problems:
 ```kotlin
val f1 = 1.0f - 0.9f
val f2 = 0.9f - 0.8f
if (f1 == f2) {
    println("Expected to enter here")
} else {
    println("But this block will be reached")
}

val flt1 = f1;
val flt2 = f2;
if (flt1.equals(flt2)) {
    println("Expected to enter here")
} else {
    println("But this block will be reached")
} 
 ```

Recommended example:

```kotlin
val foo = 1.03f
val bar = 0.42f
if (abs(foo - bar) > 1e-6f) {
    println("Ok")
} else {
    println("Not")
}
```

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Rule 4.1.2

This pull request closes #290

### CODE REVIEW

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter4/AccurateCalculationsWarnTest.kt [hunk 1]
```diff
@@ -150,4 +150,32 @@ class AccurateCalculationsWarnTest : LintTestBase(::AccurateCalculationsRule) {
...
         )
     }
+
+    @Test
+    @Tag(WarningNames.FLOAT_IN_ACCURATE_CALCULATIONS)
+    fun `should allow arithmetic operations inside abs in comparison`() {
+        lintMethod(
+                """
+                    |import kotlin.math.abs
+                    |
```
Comment (akuleshov7) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter4/AccurateCalculationsWarnTest.kt:160]: I would also add test with `==`

Comment (petertrr) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter4/AccurateCalculationsWarnTest.kt:160]: As far as I understand, `==` should not be allowed in this case and it is checked in rule 4.1.1
Or do you mean that we need a test that it is still not allowed?

Comment (petertrr) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter4/AccurateCalculationsWarnTest.kt:160]: @akuleshov7 I've added test with `==`

**Review by akuleshov7 — APPROVED** [review]
lgtm

**Pull request merged**